### PR TITLE
samples: Add sysbuild support to radio_test

### PR DESCRIPTION
**Pull request by dchat-nordic**
Add sysbuild support for radio_test sample.
Add usb option for remote_shell image.

### CODE REVIEW

**Review by peknis — APPROVED** [review]

File: samples/peripheral/radio_test/sysbuild/remote_shell/pin_fwd.dts [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
 
-/* Forwards nRF21540 EK pins to the network core. Network core requires these pins
- * to control front-end module by GPIO and SPI interfaces.
+/* Forwards nRF21540 EK and radio dfe pins to the network core. Network core requires these pins
+ * to control front-end module by GPIO and SPI interfaces and to perform the DFE functionality.
  */
 &gpio_fwd {
 	nrf21540-gpio-if {
 	    gpios = <&gpio1 7 0>,
+		<&gpio1 6 0>,
```
Comment (grochu) [inline samples/peripheral/radio_test/sysbuild/remote_shell/pin_fwd.dts:13]: Is this the ANT_SEL pin? Was it missing in the old dts file?

File: samples/peripheral/radio_test/sysbuild.conf
Comment (grochu) [inline samples/peripheral/radio_test/sysbuild.conf:1]: Is this file still needed when building just for Radio Core?

File: samples/peripheral/radio_test/prj_usb.conf [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
```
Comment (grochu) [inline samples/peripheral/radio_test/prj_usb.conf:2]: Please use the date consistently - although technically, this is not a new configuration (this configuration existed elsewhere), the file is new, introduced for sysbuild. Please align with other build system files in this sample.

File: samples/peripheral/radio_test/sysbuild.conf
Comment (dchat-nordic) [reply samples/peripheral/radio_test/sysbuild.conf:1]: The file should only affect the nrf5340_cpunet compilation. The partition manager compilation seems to dislike being invoked from cpunet first (instead from cpuapp).

In all other compilation cases this file should not have any effect.

File: samples/peripheral/radio_test/sysbuild/remote_shell/pin_fwd.dts (hunk 1 above)
Comment (dchat-nordic) [reply samples/peripheral/radio_test/sysbuild/remote_shell/pin_fwd.dts]: This is a mistake added here. This entry is not needed for radio_test.

**Review by grochu — APPROVED** [review]

File: samples/peripheral/radio_test/README.rst [hunk 3]
```diff
@@ -206,14 +206,14 @@ For example, when building on the command line, use the following command:
 
 .. code-block:: console
 
-   west build samples/peripheral/radio_test -b nrf5340dk/nrf5340/cpunet -- -DCONFIG_RADIO_TEST_USB=y
+   west build samples/peripheral/radio_test -b nrf5340dk/nrf5340/cpunet --sysbuild -- -DFILE_SUFFIX=usb
```
Comment (maje-emb) [inline samples/peripheral/radio_test/README.rst:209]: Is `--sysbuild` needed?

File: samples/peripheral/radio_test/sysbuild/remote_shell/prj_usb.conf [hunk 4]
```diff
@@ -1,13 +1,22 @@
...
 CONFIG_CONSOLE=n
 CONFIG_UART_CONSOLE=n
 
+CONFIG_SERIAL=y
+CONFIG_UART_INTERRUPT_DRIVEN=y
+CONFIG_UART_LINE_CTRL=y
+
+CONFIG_MBOX=y
+CONFIG_IPC_SERVICE=y
+CONFIG_IPC_SERVICE_BACKEND_RPMSG=y
```
Comment (maje-emb) [inline samples/peripheral/radio_test/sysbuild/remote_shell/prj_usb.conf:16]: Isn't better to use icbmsg backend?

File: samples/peripheral/radio_test/sysbuild.cmake [hunk 5]
```diff
@@ -0,0 +1,32 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+get_property(PM_DOMAINS GLOBAL PROPERTY PM_DOMAINS)
+
+# Include app core image if enabled
+if(SB_CONFIG_BOARD_NRF5340DK_NRF5340_CPUNET)
```
Comment (maje-emb) [inline samples/peripheral/radio_test/sysbuild.cmake:9]: CPUNET or CPUAPP?

Comment (dchat-nordic) [reply samples/peripheral/radio_test/sysbuild.cmake:9]: CPUNET, because if you build on nrf5340_cpunet the CPUNET board is passed to sysbuild.

File: samples/peripheral/radio_test/sysbuild/remote_shell/prj_usb.conf (hunk 4 above)
Comment (dchat-nordic) [reply samples/peripheral/radio_test/sysbuild/remote_shell/prj_usb.conf:16]: It is not tested yet on nrf5340.

File: samples/peripheral/radio_test/README.rst (hunk 3 above)
Comment (dchat-nordic) [reply samples/peripheral/radio_test/README.rst:209]: Yes to build the remote_shell subimage
